golem-worker-service-base/src/api/custom_http_request_api.rs

```diff
@@ -40,13 +38,7 @@ impl CustomHttpRequestApi {
         worker_request_executor_service: Arc<
             dyn GatewayWorkerRequestExecutor<Namespace> + Sync + Send,
         >,
-        api_definition_lookup_service: Arc<
```

Comment:
I think making the lookup polymorphic is a bit overcomplicated. I would propose having separate traits for the other input types